### PR TITLE
Split inferShapes and add verify to PRelu Op.

### PR DESCRIPTION
**Pull request by redbopo**
First pull request on onnx-mlir, mostly follow the code like Pow Op.
Maybe need to add some error test, but not quite sure.

### CODE REVIEW

File: utils/gen_onnx_mlir.py [hunk 1]
```diff
@@ -282,6 +282,7 @@
     'GlobalAveragePool',
     'GlobalMaxPool',
     'Identity',
+    'PRelu',
```
Comment (tungld) [inline utils/gen_onnx_mlir.py:285]: This should be added to `OpsWithVerifier` list instead of `OpsWithCanonicalizer` list.

**Review by tungld — COMMENTED** [review]
> Maybe need to add some error test, but not quite sure.

Yes, please add some lit tests for this.

**Comment by redbopo** [comment]
Thanks @AlexandreEichenberger  and @tungld . 
I also notice the 
`func @test_pow_verifier_1()` in invalud.mlir with no expected-error,
And need another patch to fix it.

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by tungld** [comment]
@redbopo could you resolve the conflict and fix the DCO issue? You can click on Details of DCO and follow the instruction.

> I also notice the func @test_pow_verifier_1() in invalud.mlir with no expected-error,

Thanks for finding this! It's ok to fix it here if you are interested.

**Comment by tungld** [comment]
@redbopo any progress on this? thanks!

**Comment by redbopo** [comment]
@tungld sorry for late reply. I will do a check soon.

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by redbopo** [comment]
Hello @tungld , aggreed to CLA now, and merge the main branch.
Is that only first time to contribute  , 
will need this CLA and jenkins-droid cannot auto-run ?

**Comment by AlexandreEichenberger** [comment]
I believe the first few times you need approval of folks with more privileges.

**Review by tungld — CHANGES_REQUESTED** [review]
I am sorry that I forgot to submit my review.

**Comment by tungld** [comment]
@redbopo could you please resolve the conflicts and address the above comment? Thanks!

File: src/Dialect/ONNX/ONNXOps.cpp [hunk 2]
```diff
@@ -1150,6 +1145,17 @@ LogicalResult ONNXPReluOp::inferShapes(
   return success();
 }
 
+LogicalResult ONNXPReluOp::verify() {
+  auto xShape = X().getType().cast<ShapedType>().getShape();
```
[the remark this answers is not in the document]
Comment (redbopo) [reply src/Dialect/ONNX/ONNXOps.cpp]: Fix now. @tungld

**Comment by tungld** [comment]
@redbopo I saw clang-format failed with this info:
```diff
--- ./src/Dialect/ONNX/ONNXOps.cpp	(original)
+++ ./src/Dialect/ONNX/ONNXOps.cpp	(reformatted)
@@ -1133,7 +1133,8 @@
 
 LogicalResult ONNXPReluOp::verify() {
   ArrayRef<int64_t> xShape = X().getType().cast<ShapedType>().getShape();
-  ArrayRef<int64_t> slopeShape = slope().getType().cast<ShapedType>().getShape();
+  ArrayRef<int64_t> slopeShape =
+      slope().getType().cast<ShapedType>().getShape();
 
   // PRelu supports unidirectional broadcasting, that is slope should be
   // unidirectional broadcastable to input X.
```

Please update your code accordingly. Thanks!

**Comment by tungld** [comment]
@redbopo you forgot signing DCO for some last commits, please sign and it's ready to merge. Thanks!

**Comment by redbopo** [comment]
Sorry for that. DCO success now. @tungld

**Comment by tungld** [comment]
@jenkins-droid test this please

**Review by tungld — APPROVED** [review]
LGTM. Thanks for your contribution!

**Pull request merged**